### PR TITLE
Improve test coverage for Slice test

### PR DESCRIPTION
**Pull request by sreecha**

### CODE REVIEW

**Comment by ctiller** [comment]
Looks like there's a leak in gpr_slice_test

File: include/grpc/support/slice.h [hunk 1]
```diff
@@ -175,6 +175,10 @@ gpr_slice gpr_empty_slice(void);
 int gpr_slice_cmp(gpr_slice a, gpr_slice b);
 int gpr_slice_str_cmp(gpr_slice a, const char *b);
 
+/* Returns a c-string representation of a slice. Useful for logging or debugging
+   purposes only. The string representation can change in future */
+char *gpr_slice_to_cstring(gpr_slice slice);
```
Comment (yang-g) [inline include/grpc/support/slice.h]: Do we need this? I am not sure whether we can do something like `%.*s` for gpr_log to log a slice.

File: test/core/support/slice_test.c [hunk 2]
```diff
@@ -206,6 +221,7 @@ static void test_slice_from_copied_string_works(void) {
   LOG_TEST_NAME("test_slice_from_copied_string_works");
 
   slice = gpr_slice_from_copied_string(text);
+  gpr_log(GPR_INFO, "slice: %s", gpr_slice_to_cstring(slice));
```
Comment (yang-g) [inline test/core/support/slice_test.c]: Is this log required to add coverage?

**Comment by yang-g** [comment]
The result from gpr_slice_to_cstring should be gpr_free'd?

**Comment by sreecha** [comment]
Yeah, the result should be gpr_freed.

File: test/core/support/slice_test.c (hunk 2 above)
Comment (sreecha) [inline test/core/support/slice_test.c]: Yes. gpr_slice_to_cstring function has no coverage.

File: include/grpc/support/slice.h (hunk 1 above)
Comment (sreecha) [inline include/grpc/support/slice.h]: Might be useful to have if we want to change string representation of slice in future.

Comment (sreecha) [inline include/grpc/support/slice.h]: On a second thought, maybe it is better to just remove this function. No one calls it currently and can be added in future if needed. Doesn't feel like its worth going through the pain just for code coverage purposes..

**Pull request merged**